plugins/modules/clustering/nomad/nomad_job.py

```diff
@@ -0,0 +1,291 @@
+    timeout: 120
+
+- name: Stop job
+  community.general.nomad_job:
+    host: localhost
+    state: absent
+    name: api
+
+- name: Force job to start
+  community.general.nomad_job:
+    host: localhost
+    state: present
+    name: api
+    timeout: 120
+    force_start: true
+'''
+
+
+import os
+import json
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils._text import to_native
+
+import_nomad = None
+try:
+    import nomad
+    import_nomad = True
+except ImportError:
+    import_nomad = False
+
+
+def run():
+    module = AnsibleModule(
+        argument_spec=dict(
+            host=dict(required=True, type='str'),
+            state=dict(required=True, choices=['present', 'absent']),
+            use_ssl=dict(type='bool', default=True),
+            timeout=dict(type='int', default=5),
+            validate_certs=dict(type='bool', default=True),
+            client_cert=dict(type='path', default=None),
+            client_key=dict(type='path', default=None),
+            namespace=dict(type='str', default=None),
+            name=dict(type='str', default=None),
+            content_format=dict(choices=['hcl', 'json'], default='hcl'),
+            content=dict(type='str', default=None),
+            force_start=dict(type='bool', default=False),
+            token=dict(type='str', default=None, no_log=True)
+        ),
+        supports_check_mode=True,
+        mutually_exclusive=[
+            ["name", "content"]
+        ],
+        required_one_of=[
+            ['name', 'content']
+        ]
+    )
+
+    if not import_nomad:
+        module.fail_json(msg=missing_required_lib("python-nomad"))
+
+    certificate_ssl = (module.params.get('client_cert'), module.params.get('client_key'))
+
+    nomad_client = nomad.Nomad(
+        host=module.params.get('host'),
+        secure=module.params.get('use_ssl'),
+        timeout=module.params.get('timeout'),
+        verify=module.params.get('validate_certs'),
+        cert=certificate_ssl,
+        namespace=module.params.get('namespace'),
+        token=module.params.get('token')
+    )
+
+    if module.params.get('state') == "present":
+
+        if module.params.get('name') and not module.params.get('force_start'):
+            module.fail_json(msg='For start job with name, force_start is needed')
+
+        changed = False
+        if module.params.get('content'):
+
+            if module.params.get('content_format') == 'json':
+
+                job_json = module.params.get('content')
+                try:
+                    job_json = json.loads(job_json)
+                except ValueError as e:
+                    module.fail_json(msg=to_native(e))
+                job = dict()
+                job['job'] = job_json
+                try:
+                    job_id = job_json.get('ID')
+                    if job_id is None:
+                        module.fail_json(msg="Can't revover job ID")
```

Comment:
```suggestion
                        module.fail_json(msg="Cannot retrieve job ID")
```
?

Reply:
and maybe to put info that this is `None`? who knows maybe it might be important to know for users